### PR TITLE
Upgrade to PSR18  + Httplug 2.0

### PR DESCRIPTION
**Pull request by joelwurtz**
* Add support for PSR18 and httplug 2.0
 * Fixes some cs
 * Remove support for php < 7.1

### CODE REVIEW

**Comment by joelwurtz** [comment]
Failure is only due to styleci (removed the config file), integration should be removed (but maybe existing pr still have checks ?)

Should be ready to review / merge @php-http/httplug

File: src/Client.php [hunk 1]
```diff
@@ -62,7 +63,7 @@ public function __construct(ResponseFactory $responseFactory = null, array $conf
     /**
      * {@inheritdoc}
      */
-    public function sendRequest(RequestInterface $request)
+    public function sendRequest(RequestInterface $request): ResponseInterface
```
Comment (dbu) [inline src/Client.php]: i think this is only correct for httplug 2, but not for 1. afaik only php 7.2 and newer allows to restrict the return type in an extending method, or am i wrong?

File: src/RequestWriter.php [hunk 2]
```diff
@@ -37,9 +36,8 @@ protected function writeRequest($socket, RequestInterface $request, $bufferSize
     /**
      * Write Body of the request.
      *
-     * @param resource         $socket
-     * @param RequestInterface $request
-     * @param int              $bufferSize
+     * @param resource $socket
+     * @param int      $bufferSize
```
Comment (dbu) [inline src/RequestWriter.php]: why not declare the int argument type, as we move to php 7.1 anyways?

**Comment by dbu** [comment]
i just now removed styleci hooks in this github repo

File: src/Client.php (hunk 1 above)
Comment (joelwurtz) [reply src/Client.php]: will add 7.2 in travis matrix to test this

File: composer.json [hunk 3]
```diff
@@ -9,17 +9,18 @@
         }
     ],
     "require": {
-        "php": "^5.5 || ^7.0",
+        "php": "^7.1",
         "symfony/options-resolver": "^2.6 || ^3.0 || ^4.0",
-        "php-http/httplug": "^1.0",
-        "php-http/message-factory": "^1.0.2",
+        "php-http/httplug": "^1.0 || ^2.0",
```
Comment (Nyholm) [inline composer.json]: This is not possible. We need to do 2.0 only

File: composer.json [hunk 4]
```diff
@@ -9,17 +9,18 @@
...
+        "php-http/message-factory": "^1.0.2 || ^2.0",
```
Comment (Nyholm) [inline composer.json]: Should we use PSR-17 instead?

File: composer.json [hunk 5]
```diff
@@ -9,17 +9,18 @@
...
         "php-http/discovery": "^1.0"
```
Comment (Nyholm) [inline composer.json]: Is discovery really needed?

I suggest removing it.

File: src/Client.php (hunk 1 above)
Comment (Nyholm) [reply src/Client.php]: @dbu, You are correct

File: composer.json (hunk 3 above)
Comment (Nyholm) [reply composer.json]: Unless you use the VersionClientBridge trait added in php-http/client-common 1.9 and 2.0

File: composer.json (hunk 5 above)
Comment (joelwurtz) [reply composer.json]: It's used for discovering psr17 factories (to create response)

**Comment by joelwurtz** [comment]
@dbu i thin there is still a problem somewhere in this repo settings, there is no more travis ci integration and merge request expect style ci integration to bo reported (but it should not)

**Comment by Nyholm** [comment]
Sorry, I replied to Joel in a direct message. I updated the branch protection so it does no longer require style CI to pass.

**Comment by dbu** [comment]
ah, i missed that. there is now something with travis-ci as well. its correct to require it, but it seems to not have run although i have set the callback hook in the settings. am i missing something?

File: composer.json [hunk 6]
```diff
@@ -9,17 +9,16 @@
         }
     ],
     "require": {
-        "php": "^5.5 || ^7.0",
-        "symfony/options-resolver": "^2.6 || ^3.0 || ^4.0",
-        "php-http/httplug": "^1.0",
```
Comment (dbu) [inline composer.json]: we need to require httplug 2 as the client still implements the httplug client interface. (which imho is correct to allow for smooth upgrades)

Comment (joelwurtz) [reply composer.json]: Fixed, thanks for pointing this out

**Comment by joelwurtz** [comment]
I'm still confused about settings of the repository, there is no more travis build on 2.x is this normal ?

**Pull request merged**